proxy/src/lib.rs

```diff
@@ -259,9 +259,16 @@ where
                 config.inbound_router_capacity,
                 config.inbound_router_max_idle_age,
             );
+            let tls_settings = match &config.tls_settings {
+                Some(settings) => Conditional::Some(tls::ConnectionConfig {
+                    identity: settings.service_identity.clone(),
+                    config: tls_server_config
+                }),
+                None => Conditional::None(tls::ReasonForNoTls::NoConfig),
```

Comment:
TIOLI: Seems like this is a common enough pattern that a function like 
```rust
impl<C, R> Conditional<C, R> {
    // ... 
    pub fn from_option(opt: Option<C>, reason: R) -> Conditional<C, R> {
        match opt {
            Some(c) => Conditional::Some(c),
            None => Conditional::None(reason),
        }
    }
}
```
would be nice?

(Or even an extension trait that gives options a `conditional_or()` conversion?)

Reply:
FYI, I just looked at implementing `map` and having it doesn't reduce the amount of code (in terms of number of lines), so I threw it away.